src/components/Modals/ReplaceCourseModal.vue

```diff
@@ -0,0 +1,294 @@
+            }
+          }
+        }
+      }
+    },
+    // handles the add button
+    handleAdd() {
+      this.selecting = false;
+      this.getSemestersTaken();
+      const count = this.semestersTaken.length;
+      if (count === 0) {
+        // opens the add modal if the course does not exist
+        this.needToAdd = true;
+      } else if (count > 1) {
+        // opens the duplicates modal if the course exists 2+ times
+        this.hasDuplicates = true;
+      } else {
+        // closes the modal if the course exists exactly once
+        this.closeCurrentModal();
+      }
+    },
+    selectCourse(result: CornellCourseRosterCourse) {
+      this.selectedCourse = result;
+      this.$emit('select-course', this.selectedCourse);
+      this.getReqsRelatedToCourse(result);
+      this.handleAdd();
+    },
+    closeCurrentModal() {
+      this.$emit('close-replace-course-modal');
+    },
+    getReqsRelatedToCourse(selectedCourse: CornellCourseRosterCourse) {
+      const {
+        relatedRequirements,
+        selfCheckRequirements,
+        automaticallyFulfilledRequirements,
+      } = getRelatedUnfulfilledRequirements(
+        selectedCourse,
+        store.state.groupedRequirementFulfillmentReport,
+        store.state.onboardingData,
+        store.state.toggleableRequirementChoices,
+        store.state.overriddenFulfillmentChoices,
+        store.state.userRequirementsMap
+      );
+      const automaticallyFulfilledRequirementIds = new Set(
+        automaticallyFulfilledRequirements.map(({ id }) => id)
+      );
+
+      const reqFilter = (req: RequirementWithIDSourceType) =>
+        !automaticallyFulfilledRequirementIds.has(req.id);
+
+      this.automaticallyFulfilledRequirements = automaticallyFulfilledRequirements.map(
+        ({ name }) => name
+      );
+      this.relatedRequirements = relatedRequirements.filter(reqFilter);
+      this.selfCheckRequirements = selfCheckRequirements.filter(reqFilter);
+      if (this.relatedRequirements.length > 0) {
+        this.selectedRequirementID = this.relatedRequirements[0].id;
+      } else {
+        this.selectedRequirementID = '';
+      }
+    },
+    addItem() {
+      if (this.editMode) {
+        this.editMode = false;
+      } else {
+        this.addCourse();
+      }
+    },
+    addCourse() {
+      if (this.selectedCourse == null) return;
+      this.$emit('add-course', this.selectedCourse, {
+        optOut: getRelatedRequirementIdsForCourseOptOut(
+          this.selectedCourse.crseId,
+          this.selectedRequirementID,
+          store.state.groupedRequirementFulfillmentReport,
+          store.state.toggleableRequirementChoices,
+          store.state.userRequirementsMap
+        ),
+        // Only include the selected requirement from opt-in.
+        acknowledgedCheckerWarningOptIn: this.selfCheckRequirements
+          .filter(it => it.id === this.selectedRequirementID)
+          .map(it => it.id),
+        arbitraryOptIn: {},
+      });
```

Comment:
Wouldn't this be an arbitrary opt in choice? Can you add a note to update this?

Reply:
Hmm, I understand what arbitrary opt in is, but I'm not exactly sure how to fix this?

Reply:
I would remove the acknowledged opt in and just make a note for the future, we can discuss more in person

Reply:
I think after all the merging, this may be resolved